src/Utility/ExprFunctions.C

```diff
@@ -499,6 +499,25 @@ EvalMultiFunction<ParentFunc, Ret>::evalGradient (const Vec3& X) const
 }
 
 
+template <class ParentFunc, class Ret>
+std::vector<Real>
+EvalMultiFunction<ParentFunc,Ret>::evalHessian (const Vec3& X) const
+{
+  std::vector<Real> result(this->p.size()*this->nsd*this->nsd);
+  std::vector<SymmTensor> dx;
+  for (const auto& f : this->p)
+    dx.push_back(f->hessian(X));
+
+  size_t k = 0;
+  for (size_t d2 = 1; d2 <= this->nsd; ++d2)
+    for (size_t d1 = 1; d1 <= this->nsd; ++d1)
+      for (size_t i = 0; i < this->p.size(); ++i)
+        result[k++] = dx[i](d1,d2);
```

Comment:
Guess you could use `reserve()` and `push_back()` for  the result variable to avoid the _k_ index (similar for the evalGradient method)

Reply:
done.